pages/index.js

```diff
@@ -2,21 +2,48 @@ import AddCalories from "../components/AddCalories/AddCalories";
 import useCalorieStore from "../utils/useCalorieStore";
 import styled from "styled-components";
 import ConsumedList from "../components/ConsumedList/ConsumedList";
+import { useEffect } from "react";
 
 const StyledDiv = styled.div`
   display: flex;
   flex-direction: column;
   align-items: center;
 `;
-
+const ListContainer = styled.div`
+  position: absolute;
+  top: 4rem;
+  width: 274px;
+  height: 80%;
+  left: 50%;
+  transform: translateX(-50%);
+  background: grey;
+  opacity: 0.7;
+`;
 export default function HomePage() {
   const { dailyCount } = useCalorieStore();
-
+  useEffect(() => {
+    document.addEventListener("click", () => {
+      const popUpList = document.getElementById("hiddenList");
+      popUpList.hasAttribute("hidden")
+        ? null
+        : popUpList.setAttribute("hidden", true);
+    });
+  }, []);
   return (
     <StyledDiv>
-      <h1>{dailyCount}</h1>
+      <button
+        onClick={(event) => {
+          event.stopPropagation();
+          const popUpList = document.getElementById("hiddenList");
+          popUpList.removeAttribute("hidden");
+        }}
+      >
+        {dailyCount}
+      </button>
       <AddCalories />
-      <ConsumedList />
+      <ListContainer id="hiddenList" hidden>
+        <ConsumedList />
+      </ListContainer>
```

Comment:
This is not the way you'd want to do this. You should use a state (`useState` not zustand since this state is truly local) to store weather the ListContainer is visible.

You should not be using `useEffect` and also not `getElementById`.

Try starting with something like:

```js
const [isListOpen, setIsListOpen] = useState(false);
```

and

```js
{isListOpen ? (
  <ListContainer>
      <ConsumedList />
    </ListContainer>
) : null}
```

then in your event listener toggle the state using `setIsListOpen`.

Reply:
Thx for the advice. I used useEffect because I can't make the whole document a "button" without it existing first ?